openhtf/util/logs.py

```diff
@@ -199,40 +201,48 @@ def emit(self, record):
+    super(CliFormatter, self).format(record)
+    localized_time = datetime.datetime.fromtimestamp(record.created)
+    terse_time = localized_time.strftime(u'%H:%M:%S')
+    terse_level = record.levelname[0]
+    terse_name = record.name.split('.')[-1]
+    if record.name.startswith(RECORD_LOGGER_PREFIX):
+      # The [1:] split used here is to discard the extra leading dot.
+      loggers = record.name.split(RECORD_LOGGER_PREFIX)[1][1:].split('.')
+      # If nested under a test record, show where the message came from.
+      if len(loggers) > 1:
+        terse_name = '<{type}: {identifier}>'.format(
+            type=loggers[1], # First thing after UUID should be the logger type.
+            identifier=loggers[-1]) # Last thing should uniquely id the logger.
+      else:
+        # Fall back to using the last five characters of the test UUID.
+        terse_name = '<test %s>' % loggers[0][-5:]
+    return '{lvl} {time} {logger} - {msg}'.format(lvl=terse_level,
+                                                     time=terse_time,
+                                                     logger=terse_name,
+                                                     msg=record.message)
+
+
 @functions.call_once
-def setup_logger():
-  """Configure logging for OpenHTF."""
-  record_logger = logging.getLogger(RECORD_LOGGER)
-  record_logger.propagate = False
-  record_logger.setLevel(logging.DEBUG)
-
-  # TODO: make printed timestamp optional?
-  record_console_handler_formatter = logging.Formatter(
-    u'[%(asctime)s]    %(message)s', '%H:%M:%S')
-  record_console_handler = logging.StreamHandler(stream=sys.stdout)
-  record_console_handler.setFormatter(record_console_handler_formatter)
-  record_console_handler.setLevel(DEFAULT_RECORD_VERBOSITY.upper())
-  record_logger.addHandler(record_console_handler)
-
-  logger = logging.getLogger(LOGGER_PREFIX)
-  logger.propagate = False
-  logger.setLevel(logging.DEBUG)
-  formatter = logging.Formatter(u'%(asctime)s - %(levelname)s - %(message)s')
-  if LOGFILE:
-    try:
-      cur_time = str(util.time_millis())
-      file_handler = logging.FileHandler('%s.%s' % (LOGFILE, cur_time))
-      file_handler.setFormatter(formatter)
-      file_handler.setLevel(DEFAULT_LOGFILE_LEVEL.upper())
-      file_handler.addFilter(MAC_FILTER)
-      logger.addHandler(file_handler)
-    except IOError as exception:
-      print('Failed to set up log file due to error: %s. '
-             'Continuing anyway.' % exception)
-
-  if not QUIET:
-    console_handler = logging.StreamHandler(stream=sys.stderr)
-    console_handler.setFormatter(formatter)
-    console_handler.setLevel(DEFAULT_LEVEL.upper())
-    console_handler.addFilter(MAC_FILTER)
-    logger.addHandler(console_handler)
+def configure_cli_logging():
+  """Configure OpenHTF to log to the CLI based on verbosity arg."""
+  args, _ = ARG_PARSER.parse_known_args()
+  if not args.v: # The default behavior is not to log anything to the CLI.
```

Comment:
When I run a test that raises an error (e.g. add `raise RuntimeError` to hello_world.py) I get a stack trace in the console (without -v). Same if I abort before DUT ID is set. Seems unintended?

I'm also surprised that without -v, I don't see in the console whether the test passed or failed (again using hello_world.py). Is that intended?

Reply:
I must be missing something with stack traces. I'll take another look. The end test result thing I could go either way on. I can add a (colorized!) cli_print of it at the end of the test.